### PR TITLE
Fix gists fetch through ajax/json

### LINKED ISSUE #3519: gist oEmbeds broken
#### What I expected
When I embed one specific file from a gist (e.g. `[gist file="create_date_column_for_issues.php"]https://gist.github.com/CFXd/8e7927848100d4cd4d6d[/gist]`) then the gist should output and embed.

#### What happened instead
Resulting markup from the above example:

```html
<style>.gist table { margin-bottom: 0; }</style>
<div class="gist-oembed" data-gist="CFXd/8e7927848100d4cd4d6d.json?file=create_date_column_for_issues.php"></div>
```

#### Steps to reproduce the issue
Follow the example above on any gist with multiple files and specify one file in your embed code.

#### Last known working version
3.7

#### Interesting to note
If I simply enter the gist URL in my post content then all of the previous gist with file arguments embeds work, but it then also displays the entire gist's files yet again.

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #3519.

The gist shortcode wasn't working correctly after #2857 was merged. This PR enqueues the JavaScript file that fetches the gist using ajax to make them work fine. It removes the now unneeded enqueue call from embed handler since file is enqueued in shortcode callback.

Additionally, itr introduces support for embedding a specific file in a gist, working both in editor preview and front end.

Finally, it fixes the code so it follows WP standards and adds inline documentation for functions.

### CODE REVIEW

File: modules/shortcodes/gist.php [hunk 1]
```diff
@@ -1,58 +1,91 @@
...
+ *
+ * @since 3.9.0
+ *
+ * @param string $id The ID of the gist.
+ *
+ * @return string
+ */
+function github_gist_simple_embed( $id ) {
+	$id = str_replace( 'json', 'js', $id );
+	return '<script type="text/javascript" src="//gist.github.com/' . $id . '"></script>';
```
Comment (kraftbj) [inline modules/shortcodes/gist.php]: `//` is an anti-patten now, what about just using `https`? All `http` pages on gist.github.com redirect anyhow.

Comment (CFXd) [inline modules/shortcodes/gist.php]: I haven't been able to confirm whether your suggestion solves my latest problem https://github.com/Automattic/jetpack/issues/3519#issuecomment-194472454.

Comment (eliorivero) [inline modules/shortcodes/gist.php]: @kraftbj good catch, thanks, it's fixed now.

@CFXd I'll be checking the issue with ssl asap and will let you know.

**Comment by CFXd** [comment]
I tried the latest changes and it still does not display Gists on an SSL enabled site. Reverting back to 3.7 works fine on SSL fwiw.

**Comment by eliorivero** [comment]
User confirms it works so it's good to go into review https://github.com/Automattic/jetpack/issues/3519#issuecomment-195624287

**Comment by zinigor** [comment]
Tested shortcodes and oEmbeds, everything seems to be working fine even with IS, where we previously had problems. Thanks!

**Pull request merged**